sale_delivery_date/models/sale_order_line.py

```diff
@@ -136,97 +213,33 @@ def _get_delays(self):
         return customer_lead, security_lead, workload
 
     def _expected_date(self):
-        # Computes the expected date with respect to the WH calendar, if any.
-        expected_date = super()._expected_date()
-        expected_date = self._cutoff_time_delivery_expected_date(expected_date)
-        expected_date = self._warehouse_calendar_expected_date(expected_date)
-        expected_date = self._delivery_window_expected_date(expected_date)
-        return expected_date
-
-    def _warehouse_calendar_expected_date(self, expected_date):
-        calendar = self.order_id.warehouse_id.calendar_id
-        if calendar:
-            customer_lead, security_lead, workload = self._get_delays()
-            td_customer_lead = timedelta(days=customer_lead)
-            td_security_lead = timedelta(days=security_lead)
-            # plan_days() expect a number of days instead of a delay
-            workload_days = self._delay_to_days(workload)
-            # Remove customer_lead added to order_date in sale_stock
-            expected_date -= td_customer_lead
-            # Add the workload, with respect to the wh calendar
-            expected_date = calendar.plan_days(
-                workload_days, expected_date, compute_leaves=True
-            )
-            # add back the security lead
-            expected_date += td_security_lead
-        return expected_date
-
-    def _delivery_window_expected_date(self, expected_date):
+        # Overwritten to compute the expected date with respect to:
+        #   - the warehouse or partner cutoff time
+        #   - the warehouse calendar
+        #   - the delivery time window of the customer
+        customer_lead, security_lead, workload = self._get_delays()
+        date_planned = (
+            (self.order_id.date_order or fields.Datetime.now())
+            + timedelta(days=customer_lead or 0.0)
+            - timedelta(days=security_lead)
+        )
```

Comment:
According to me, the leads should be removed
```suggestion
        date_planned = self.order_id.date_order or fields.Datetime.now()
```
workload = customer_lead - security_lead
So you apply twice the workload, once here and once below taking care of the calendar.